util/difgen/gen_dif_listing.py

```diff
@@ -109,19 +110,53 @@ def _get_dif_function_info(compound, file_id):
         func_info["name"] = _get_text_or_empty(m, "name")
         func_info["prototype"] = _get_text_or_empty(
             m, "definition") + _get_text_or_empty(m, "argsstring")
-        func_info["description"] = _get_text_or_empty(m,
+        func_info["description"] = _get_html_or_empty(m,
                                                       "briefdescription/para")
 
         functions.append(func_info)
 
     return functions
 
 
-def _get_text_or_empty(element, xpath):
+def _get_html_or_empty(element: ET.Element, xpath: str) -> str:
+    """ Get a minimal HTML-rendering of the children in an element.
+
+    element is expected to be a docCmdGroup according to the DoxyGen schema [1],
+    but only a very minimal subset of formatting is transferred to semantic
+    HTML. However, the tag structure is retained by transforming all tags into
+    HTML `span` tags with a class attribute `doxygentag-ORIGINALTAGNAME`. This
+    can be used to write CSS targeting specific Doxygen tags and recreate the
+    intended formatting.
+
+    In addtion, the following semantic transformations are performed:
+    - `computeroutput` is transformed to `code`
+
+    [1] https://github.com/doxygen/doxygen/blob/master/templates/xml/compound.xsd"""
     inner = element.find(xpath)
     # if the element isn't found, return ""
     if inner is None:
         return ""
 
-    # If there's no inner text, return ""
-    return inner.text or ""
+    # Avoid modifying the passed element argument.
+    inner_copy = copy.deepcopy(inner)
+
+    for c in inner_copy.iter():
+        c.set('class', 'doxygentag-' + c.tag)
+        if c.tag == 'computeroutput':
+            c.tag = 'code'
+        else:
+            c.tag = 'span'
+
+    # Create a string from all subelements
+    text = ET.tostring(inner_copy, encoding="unicode", method="html")
+    print(text)
+    return text or ""
+
+
+def _get_text_or_empty(element: ET.Element, xpath: str) -> str:
+    """ Get all text of an element, without any tags """
+    inner = element.find(xpath)
+    if inner is None:
+        return ""
+
+    return ' '.join([element.text for element in inner.iter()]) or ""
```

Comment:
```suggestion
    return ' '.join([e for e in inner.itertext()]) or ""
```
I know I had a problem with the iteration and just did a quick check. `element.text` can be of type `NoneType`, which will cause an error when `str` is expected.
And I think `itertext()` handled all the sublements better (the implementation differs, but I can not really tell which is doing what) so even though we do not have subelements in the elements which are processed by this function, using `itertext()` might work as well?